jsengine-android/jsengine/src/main/assets/webrequest.js

```diff
@@ -24,29 +24,33 @@ var webRequest = {
           }
         },
 
-        _trigger(requestInfo) {
+        _trigger: function(requestInfo) {
           // getter for request headers
           requestInfo.getRequestHeader = function(header) {
             return requestInfo.requestHeaders[header];
           };
-          for (let listener of this.listeners) {
-              const {fn, filter, extraInfo} = listener;
-              const blockingResponse = fn(requestInfo);
-              if (blockingResponse && Object.keys(blockingResponse).length > 0) {
-                return blockingResponse;
-              }
-          }
-          return {};
+          var returnBlockingResponse = {};
+          this.listeners.forEach(function(listener) {
```

Comment:
I think using a forEach here makes the logic a little messy. I imagine the reason for the change is a `for .. of` loop, but we can use a different kind of for loop instead, which will then look closer to the original implementation:

```
for (var i=0; i < this.listeners.length; i++) {
  const listener = this.listeners[i];
  ...
}
```